src/koans/15_destructuring.clj

```diff
@@ -8,37 +8,40 @@
 
 (meditations
   "Destructuring is an arbiter: it breaks up arguments"
-  (= __ ((fn [[a b]] (str b a))
+  (= ":bar:foo" ((fn [[a b]] (str b a))
          [:foo :bar]))
 
   "Whether in function definitions"
   (= (str "An Oxford comma list of apples, "
           "oranges, "
           "and pears.")
-     ((fn [[a b c]] __)
+     ((fn [[a b c]] (str "An Oxford comma list of " a ", " b ", and " c "."  ))
       ["apples" "oranges" "pears"]))
 
   "Or in let expressions"
   (= "Rich Hickey aka The Clojurer aka Go Time aka Lambda Guru"
      (let [[first-name last-name & aliases]
            (list "Rich" "Hickey" "The Clojurer" "Go Time" "Lambda Guru")]
-       __))
+       (str first-name " " last-name ((fn [[a b c]] (str " aka " a " aka " b " aka " c)) aliases))))
 
   "You can regain the full argument if you like arguing"
   (= {:original-parts ["Stephen" "Hawking"] :named-parts {:first "Stephen" :last "Hawking"}}
      (let [[first-name last-name :as full-name] ["Stephen" "Hawking"]]
-       __))
+       {:original-parts full-name
+        :named-parts
+          {:first first-name :last last-name}
+        }))
 
   "Break up maps by key"
   (= "123 Test Lane, Testerville, TX"
      (let [{street-address :street-address, city :city, state :state} test-address]
-       __))
+       (str street-address ", " city ", " state)))
 
   "Or more succinctly"
   (= "123 Test Lane, Testerville, TX"
-     (let [{:keys [street-address __ __]} test-address]
-       __))
+     (let [{:keys [street-address city :state]} test-address]
+       (str street-address ", " city ", " state)))
 
   "All together now!"
   (= "Test Testerson, 123 Test Lane, Testerville, TX"
-     (___ ["Test" "Testerson"] test-address)))
+     (#(str (first %1) " " (last %1) ", " (%2 :street-address) ", " (%2 :city) ", " (%2 :state))["Test" "Testerson"] test-address)))
```

Comment:
Point of this exercise is deconstruction so something like:
```
(fn [[first-name last-name] {:keys [street-address city state]}]
       (str first-name " " last-name ", " street-address ", " city ", " state)
```
 would be a much better aproach.

Reply:
I though of doing it like this, but the placing of __ tricked me in to thinking I should use only that space. Maybe we could do a small comment on the code.

Reply:
It dose not work with fn as the result of fn is the compilation of the function, thus returning false. However we could do it like this:

```
 (= "Test Testerson, 123 Test Lane, Testerville, TX"
     (let [[first-name last-name] ["Test" "Testerson"] 
           {:keys [street-address city :state]} test-address], 
        (str first-name " " last-name ", " street-address ", " city ", " state)))
```